distiller/data/etls/extract_pdf.py

```diff
@@ -0,0 +1,54 @@
+import json
+import sys
+import smart_open
+import io
+from datetime import datetime
+
+from django.conf import settings
+
+from ..models.single_audit_db import Audit
+from ..models.pdf_extract import PDFExtract
+
+from ...gateways import files
+from ...extraction import nlp, pdf_utils
+from ...extraction.analyze import analyze
+from ...fac_scraper.models import FacDocument
+
+
+def setup():
+    return nlp.setup()
+
+
+def get_all_pdfs():
+    return FacDocument.objects.all().values_list('pk', flat=True)
+
+
+def process_audit_pdf(processor, pdf_id):
+    try:
+        document = FacDocument.objects.get(id=pdf_id)
+        pdf = files.input_file(f"{settings.FAC_DOCUMENT_ROOT}/{document.file_name}", mode='rb')
+        errors = pdf_utils.errors(pdf)
+        if errors:
+            sys.stdout.write(f'Could not read file: {errors}. Bailing out.\n')
+            sys.stdout.flush()
+            return
+
+        audit_results = analyze(processor, pdf)
+        for result in audit_results:
+            audit_num = result["audit"]
+            page_number = result["page_number"]
+            finding_data = result["finding_data"]
+            cap_data = result["cap_data"]
+            sys.stdout.write(f'Found audit {audit_num} on page {page_number}.\n')
+            sys.stdout.flush()
+            PDFExtract(audit_year=document.audit_year,
+                       dbkey=document.dbkey,
+                       finding_ref_nums=audit_num,
+                       finding_text=json.dumps(finding_data),
+                       cap_text=json.dumps(cap_data),
+                       last_updated=datetime.now(),
+            ).save()
```

Comment:
Just a thought - Since it's so expensive to process these files, there might be value in writing this to S3 first, so we can load the processed data from there if we need to recreate the database.

Reply:
Yep, completely agree, we can use something like pickle to write the dict to s3 (modulo security issues, have to think about it). Haven't gotten that far yet.